test/suite/test_verbose01.py

```diff
@@ -63,9 +60,12 @@ def expect_verbose(self, categories, patterns):
         # to ensure we've only generated verbose messages for the expected categories.
         verbose_messages = output.splitlines()
 
-        # If no categories were passed, we aren't expecting any output.
-        if len(categories) == 0:
+        # Check if we are expecting any output. We should not expect anything if no categories were
+        # passed.
+        if len(categories) == 0 or not expect_output:
```

Comment:
I think we can just drop `len(categories) == 0`, and just rely on `expect_output`. Should simplify the logic a bit.

Reply:
It will be easier to understand indeed, fixed in b98c5dce17865b46e102527fddc7b85d2d2a7673.